helpers/extensions/__init__.py

```diff
@@ -232,19 +235,45 @@ def build_lib_arrays(command, libname):
 
 ####
 # FORTRAN EXTENSIONS
+#
+# Note that NumPy's distutils will replace flags from the environment
+# rather than append them - e.g. see
+# https://github.com/numpy/numpy/issues/1171
+# which is taken from the discussion of the issue at
+# https://groups.google.com/a/continuum.io/forum/#!topic/anaconda/Xw57CjIcBIU
+#
+# This causes problem's building with conda's gfortran (version 7.2.0),
+# and one suggested workaround is to just expliticly add in the
+# "-shared" which is needed to build the Fortran extensions.
+#
+# It is not clear to DJB what exactly is going on under the hood
+# here, so this feels rather fragile, and system specific. Is there
+# a way to find out what the link flags would be without LDFLAGS
+# and then ensure they are included? At present this just assumes
+# that adding in -shared is enough.
+#
 ####
+
+f77_ldflags = os.environ.get('LDFLAGS', '')
```

Comment:
Before I forget, this seems like a hack, and it only solves one very specific problem on RTD. Other than that, the proper way of doing it is to change the relative option in `setup.cfg`, which is not a problem in a minimally decent build environment, as demonstrated by the fact that the CIAO build scripts, the Travis jobs configuration, the conda recipes being run on the Gitlab mirror or locally, etc. they all directly change the `setup.cfg` configuration for controlling the build.

Reply:
Actually, it also "solves" the problem of the user who is trying to build the code with a recent conda package, without having to manually adjust the configuration. Or at least that is one of the use cases that I had for this commit.

Whether it's a good or workable solution is another matter, but I want to point out it's not just a Read-The-Docs issue.